### PR TITLE
Documentation for net-shell

### PR DESCRIPTION
**Pull request by jukkar**
Note that this PR has been built on top of #13616 (so the first commit is from that PR).

### CODE REVIEW

File: doc/reference/networking/net_shell.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.. _net_shell:
+
+Network Shell
+#############
+
+Overview
+********
+
+Network shell provides helpers for figuring out network statuses,
```
Comment (dbkinder) [inline doc/reference/networking/net_shell.rst]:
```suggestion
Network shell provides helpers for figuring out network status,
```

File: doc/reference/networking/net_shell.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+enabling/disabling features and issuing commands like ping or DNS resolving.
```
Comment (dbkinder) [inline doc/reference/networking/net_shell.rst]:
```suggestion
enabling/disabling features, and issuing commands like ping or DNS resolving.
```

**Review by nashif — APPROVED** [review]

File: doc/reference/networking/net_shell.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Note that ``net-shell`` should probably not be used in production code
+as it will require extra memory. See also :ref:`generic shell <shell_label>`
+for detailed shell information.
+
+Following commands are implemented:
```
Comment (dbkinder) [inline doc/reference/networking/net_shell.rst]:
```suggestion
The following net-shell commands are implemented:
```

File: doc/reference/networking/net_shell.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+.. csv-table:: net-shell commands
+   :header: "Command", "Description"
+   :widths: auto
+
+   "net allocs", "Print network memory allocations. Only available if
+   :option:`CONFIG_NET_DEBUG_NET_PKT_ALLOC` is set."
+   "net arp", "Print information about IPv4 ARP cache. Only available if
+   :option:`CONFIG_NET_ARP` is set in IPv4 enabled networks."
+   "net conn", "Print information about network connections."
+   "net dns", "Show how DNS is configured. The command can also used to
```
Comment (dbkinder) [inline doc/reference/networking/net_shell.rst]:
```suggestion
   "net dns", "Show how DNS is configured. The command can also be used to
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
The commit message should also indicate this PR reorganizes the network API documentation (in addition to filling out more net-shell documentation).

**Comment by jukkar** [comment]
Updated according to comments.

**Review by dbkinder — APPROVED** [review]

**Pull request merged**